java/src/main/java/com/datasqrl/ai/models/openai/OpenAIChatSession.java

```diff
@@ -19,30 +26,32 @@ public class OpenAIChatSession extends AbstractChatSession<ChatMessage, ChatFunc
   OpenAITokenCounter tokenCounter;
 
   public OpenAIChatSession(ChatModel model, ChatMessage systemMessage,
-      FunctionBackend backend,
-      Map<String, Object> sessionContext) {
+                           FunctionBackend backend,
+                           Map<String, Object> sessionContext) {
     super(backend, sessionContext, null);
     this.chatModel = model;
     this.tokenCounter = OpenAITokenCounter.of(model);
     this.systemMessage = convertMessage(systemMessage);
   }
 
-  public ChatCompletionRequest.ChatCompletionRequestBuilder setContext(ChatCompletionRequest.ChatCompletionRequestBuilder requestBuilder) {
+  public List<ChatMessage> getMessages() {
```

Comment:
The reason I originally implemented this via a single setter is because I was trying to avoid calling getWindow() twice since all the token counting could potentially be expensive.
I agree that breaking the methods out leads to better encapsulation since we don't have to pass the chatcompletion request builder in here which is model specific.
Maybe consolidate the two functions into one? getMessagesAndFunctions which returns one object that contains both (as a simple @Value class)?

Reply:
OK, makes sense. But wouldn't it be easier if we wrapped these two return types into something like `Context` ? I find the way the `sessionContext` is defined and handled internally a bit hard to understand.